### PR TITLE
feat: measure scratchpad store

### PR DESCRIPTION
**Pull request by crepererum**
This at least measures how much data we have within the scratchpad, so we can see if they contribute to OOMs at the moment.

This does NOT limit the scratchpad size. I have a rough idea how I would implement that (namely assuming that input and output file sizes are the same and with some clever semaphore usage that avoids any stalling / deadlocks). I'll do this afterwards when it turns out that we use too much memory for this buffer.

Ref https://github.com/influxdata/idpe/issues/17125 .

### CODE REVIEW

**Review by domodwyer — APPROVED** [review]
Instrumentation decorators ftw

**Review by NGA-TRAN — COMMENTED** [review]
Nice.Thanks

File: compactor2/src/object_store/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+impl MetricsStore {
+    /// Create new metrics wrapper.
+    pub fn new(inner: Arc<DynObjectStore>, registry: &Registry, name: &'static str) -> Self {
+        Self {
+            inner,
+            gauge_bytes: registry
+                .register_metric::<U64Gauge>(
+                    "iox_compactor_store_bytes",
+                    "Number of bytes in the given store",
```
Comment (cannonpalms) [inline compactor2/src/object_store/metrics.rs:37]: Is this the total number of bytes in object store?

Comment (cannonpalms) [reply compactor2/src/object_store/metrics.rs:37]: It looks like the total number of bytes "written" (in memory) into the scratchpad. Sorry, wasn't super clear. Cool!

**Pull request merged**